diff --git a/node_modules/metro-file-map/src/worker.js b/node_modules/metro-file-map/src/worker.js
index 68885ba..6b4b141 100644
--- a/node_modules/metro-file-map/src/worker.js
+++ b/node_modules/metro-file-map/src/worker.js
@@ -31,7 +31,18 @@ async function worker(data) {
     data;
   const getContent = () => {
     if (content == null) {
-      content = fs.readFileSync(filePath);
+      const delays = [100, 200, 400, 800, 1000, 1000, 1000, 1000, 1000, 1000];
+      for (let i = 0; i <= delays.length; i++) {
+        try {
+          content = fs.readFileSync(filePath);
+          break;
+        } catch (err) {
+          if (err.code !== 'EBUSY' || i === delays.length) {
+            throw err;
+          }
+          Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, delays[i]);
+        }
+      }
     }
     return content;
   };
